src/rogue/protocols/packetizer/ControllerV1.cpp

```diff
@@ -183,37 +183,38 @@ void rpp::ControllerV1::applicationRx ( ris::FramePtr frame, uint8_t tDest ) {
    lUser = (frame->getFlags() >> 8) & 0xFF;
    tId   = (frame->getFlags() >> 16) & 0xFF;
 
-   for (x=0; x < frame->getCount(); x++ ) {
+   segment = 0;
+   for (it=frame->beginBuffer(); it != frame->endBuffer(); ++it) {
       ris::FramePtr tFrame = ris::Frame::create();
-      buff = frame->getBuffer(x);
 
       // Adjust header and tail, before setting new size
-      buff->adjustHeader(-8);
-      buff->adjustTail(-1);
+      (*it)->adjustHeader(-8);
+      (*it)->adjustTail(-1);
       
       // Make payload one byte longer
-      buff->adjustPayload(1);
+      (*it)->adjustPayload(1);
 
-      size = buff->getPayload();
-      data = buff->getPayloadData();
+      size = (*it)->getPayload();
+      data = (*it)->begin();
 
       data[0] = ((appIndex_ % 16) << 4);
       data[1] = (appIndex_ / 16) & 0xFF;
 
-      data[2] = x % 256;
-      data[3] = (x % 0xFFFF) / 256;
-      data[4] = x / 0xFFFF;
+      data[2] = segment % 256;
```

Comment:
Using a "%" operator is not very CPU efficient and hurts us in streaming performance. 
I would recommend implementing with bit operators instead (see what I did in the ControllerV2.cpp code).

Reply:
Ok. That can be addressed in the current pre-release branch.